string_permutation_problem/permute.py

```diff
@@ -0,0 +1,25 @@
+def permute(s):
+    out = []
+    
+    # Base Case
+    if len(s) == 1:
+        out = [s]
+        
+    else:
+        # For every letter in string
+        for i, let in enumerate(s):
+            
+            # For every permutation resulting from Step 2 and 3 described above
+            for perm in permute(s[:i] + s[i+1:]):
+                
+                # Add it to output
+                out += [let + perm]
+
+    return out
+   
```

Comment:
Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `python spacing`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpb1s7hhaw/string_permutation_problem/permute.py
+++ b/tmp/tmpb1s7hhaw/string_permutation_problem/permute.py
@@ -16,7 +16,7 @@
                 out += [let + perm]
 
     return out
-   
+
 def main():
     permute('abc')
 
```